gravitee-node-api/src/main/java/io/gravitee/node/api/secrets/runtime/discovery/DefinitionSecretRefsFinder.java

```diff
@@ -0,0 +1,33 @@
+package io.gravitee.node.api.secrets.runtime.discovery;
+
+import java.util.Map;
+
+/**
+ * Finds locations (fields/configurations) that can contain secret references and notifies a {@link DefinitionSecretRefsListener}
+ * @param <T> the type of definition
+ * @author Benoit BORDIGONI (benoit.bordigoni at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public interface DefinitionSecretRefsFinder<T> {
+    /**
+     * Compute if the <code>definition</code> passed can be processed
+     * @param definition the definition to test
+     * @return true if it can handle the definition
+     */
+    boolean canHandle(Object definition);
+
+    /**
+     * Return a description of the definition to be used when checking ACLs
+     * @param definition the definition object to analyse
+     * @param metadata additional properties related to the definition
+     * @return a descriptor of the definition
+     */
+    DefinitionDescriptor getDefinitionLocation(T definition, Map<String, String> metadata);
```

Comment:
the method and the return object are not aligned: location vs descriptor, which makes the code more difficult to understand.